frontend/src/components/welcome-page/Login.js

```diff
@@ -0,0 +1,249 @@
+import React, { useState, useEffect } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+// import { API_URL } from "../../utils/urls";
+// import { user } from "../../reducers/user"
+import { user, userAccess } from '../../reducers/user';
+
+const Login = () => {
+  //----------- LOCAL STATES ----------//
+  const [name, setName] = useState('');
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+  const [mode, setMode] = useState('signup');
+
+  //----------- SELECTORS ----------//
+  const accessToken = useSelector((store) => store.user.accessToken);
+  const errorMessage = useSelector((store) => store.user.error);
+
+  //--------- DISPATCHES ----------//
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+
+  //--------- USEEFFECT FOR THE ACCESSTOKEN ----------//
+  useEffect(() => {
+    // This condition can be only true if the user has signed up first as the accessToken is generated when a newUser model is created in the database.
+    // Once the user has signed up, the accessToken is sent by the backend in the data response and updated in the redux store by the actions.
+    if (accessToken) {
+      navigate('/dashboard');
+    }
+  }, [accessToken, navigate]);
+
+  //--------- USEEFFECT FOR THE ERRORMESSAGE ----------//
+
+  useEffect(() => {
+    // This condition can be only true if the the data comming from the database when fetching is not success.
+    // If the user tries to sign in without having signed up first or if the username does not match the password, this function executes.
+    if (errorMessage) {
+      alert(errorMessage);
+    }
+  }, [errorMessage]);
+
+  const onFormSubmit = (event, name, username, password, mode) => {
+    event.preventDefault();
+    dispatch(userAccess(name, username, password, mode));
+  };
+
+  return (
+    <LoginContainer>
+      <Wrapper>
+        {mode === 'signin' ? (
+          <LinkText>
+            <PText>First time here? </PText>
+            <Div>
+              <PButton onClick={() => setMode('signup')}>Sign Up</PButton>
+            </Div>
+          </LinkText>
+        ) : (
+          <LinkText>
+            <PText>Already have an account? </PText>
+            <Div>
+              <PButton
+                onClick={() => {
+                  setMode('signin');
+                  dispatch(user.actions.setError(null));
+                  setUsername('');
+                  setPassword('');
+                }}
```

Comment:
It's a little bit too much logic to place directly in JSX - would be great to create separate function and pass it here instead